api/api_public_blockchain.go

```diff
@@ -311,12 +282,26 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	// this makes sure resources are cleaned up.
 	defer cancel()
 
-	// Add gas fee to sender for estimating gasLimit/computing cost or calling a function by insufficient balance sender.
-	state.AddBalance(msg.ValidatedSender(), new(big.Int).Mul(new(big.Int).SetUint64(msg.Gas()), msg.GasPrice()))
-	// Get a new instance of the EVM.
-	evm, vmError, err := b.GetEVM(ctx, msg, state, header, vmCfg)
+	// TODO-Klaytn: Klaytn is using fixed baseFee as now but, if we change this fixed baseFee as dynamic baseFee, we should update this logic too.
+	fixedBaseFee := new(big.Int).SetUint64(params.BaseFee)
+	intrinsicGas, err := types.IntrinsicGas(args.Data, nil, args.To == nil, b.ChainConfig().Rules(header.Number))
+	if err != nil {
+		return nil, 0, 0, 0, err
+	}
+	msg, err := args.ToMessage(globalGasCap.Uint64(), fixedBaseFee, intrinsicGas)
```

Comment:
Can this logic handle the case where the optional values are not defined in args?

Reply:
Yes, it can. It checks all variables and params at args are not nil, or preset.